// ng-rating-bar/esm2020/lib/ng-rating-bar.module.mjs
import { NgModule } from '@angular/core';
import { NgRatingBarComponent } from './ng-rating-bar.component';
import { CommonModule } from '@angular/common';
import * as i0 from "@angular/core";
export class NgRatingBarModule {
}
NgRatingBarModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.1.1", ngImport: i0, type: NgRatingBarModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NgRatingBarModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.1.1", ngImport: i0, type: NgRatingBarModule, declarations: [NgRatingBarComponent], imports: [CommonModule], exports: [NgRatingBarComponent] });
NgRatingBarModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.1.1", ngImport: i0, type: NgRatingBarModule, imports: [CommonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.1.1", ngImport: i0, type: NgRatingBarModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        NgRatingBarComponent
                    ],
                    imports: [
                        CommonModule
                    ],
                    exports: [
                        NgRatingBarComponent
                    ]
                }]
        }] });
